src/core/ext/xds/xds_api.h

```diff
@@ -595,12 +595,8 @@ class XdsApi {
   };
   using ResourceMetadataMap =
       std::map<absl::string_view /*resource_name*/, const ResourceMetadata*>;
-  struct ResourceTypeMetadata {
-    absl::string_view version;
-    ResourceMetadataMap resource_metadata_map;
-  };
   using ResourceTypeMetadataMap =
-      std::map<absl::string_view /*type_url*/, ResourceTypeMetadata>;
+      std::map<absl::string_view /*type_url*/, ResourceMetadataMap>;
```

Comment:
Without the per-resource version field, I think we might not need a map to communicate between xds_client and xds_api. What do you think about using a list of resource metadata instead? In this way, the data structure will be more similar to the target generic_xds_config. And we might need to add a "type_url" field in resource metadata.

Reply:
That seems like a less memory-efficient data structure, since it would unnecessarily repeat the type URL for every entry.